CSC307Fall2018/team-too
# Add Inventory and Item Page

# Summary
Adds an inventory page for staff. The page has filtering capabilities via the categories. Staff can add a product by clicking on "Add Item" and update or delete a product by first clicking on an individual item ID. Within this "Item" page, one can edit a product's fields and delete a product. All input requires generic validation. Changing a product in any way (i.e. CRUD) should see immediate changes upon being redirected to the inventory page. 

# Missing Features
Categories for products are not yet implemented into the CRUD operations. They cannot be changed for now. 

# QA
- In the inventory page (accessed manually via /admin/inventory) check to see that filtering categories works as expected (i.e. selecting a category only shows products of that category, no category selected shows all products, etc.)
- Click on "Add Item" to see that it brings you to an unfilled Item page. This page will not have a "Delete" button. Verify that the input is required and leaving any empty prompts you to fill it in. This is not the case for "Cancel". That should redirect only.
- In the inventory page, verify that selecting a product's ID leads you to an "Item" page with the appropriate product information.
- Verify that, on a populated "Item" page, you can edit fields and upon clicking "Save Item" you are redirected to "Inventory" and see the changes immediately. Same goes for clicking "Delete Item".

Closes #48 
Closes #101
Closes #22 
Closes #21 
Closes #20
